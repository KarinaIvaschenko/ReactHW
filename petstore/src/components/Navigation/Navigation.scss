.navigation {
    margin: 0 auto;
}
.navigation__list {
    display: flex;
    padding-left: 0;
    list-style: none;
    gap: 100px;
}
.navigation__link {
    text-decoration: none;
    font-size: 25px;
    &.active {
        color: black;
    }
}
