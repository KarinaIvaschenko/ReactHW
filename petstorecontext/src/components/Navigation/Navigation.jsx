import { NavLink } from "react-router-dom";
import "./Navigation.scss";

const Navigation = () => {
    return (
        <nav className="navigation">
            <ul className="navigation__list">
                <li className="navigation__item">
                    <NavLink className="navigation__link" to="/">
                        Home
                    </NavLink>
                </li>
                <li className="navigation__item">
                    <NavLink className="navigation__link" to="/favorites">
                        Favorites
                    </NavLink>
                </li>
                <li className="navigation__item">
                    <NavLink className="navigation__link" to="/cart">
                        Cart
                    </NavLink>
                </li>
            </ul>
        </nav>
    );
};

export default Navigation;
