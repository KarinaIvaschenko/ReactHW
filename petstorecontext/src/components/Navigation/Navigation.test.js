import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import Navigation from "./Navigation";
import { BrowserRouter as Router } from "react-router-dom";

describe("Navigation", () => {
    test("We have Navigation?", () => {
        const { container } = render(
            <Router>
                <Navigation />
            </Router>
        );
        expect(container).toBeInTheDocument();
    });
    test("Snapshot Navigation", () => {
        const { container } = render(
            <Router>
                <Navigation />
            </Router>
        );
        expect(container).toMatchSnapshot();
    });
});
