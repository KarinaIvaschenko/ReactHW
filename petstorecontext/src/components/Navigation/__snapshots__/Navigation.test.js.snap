// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Snapshot Navigation 1`] = `
<div>
  <nav
    class="navigation"
  >
    <ul
      class="navigation__list"
    >
      <li
        class="navigation__item"
      >
        <a
          aria-current="page"
          class="navigation__link active"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="navigation__item"
      >
        <a
          class="navigation__link"
          href="/favorites"
        >
          Favorites
        </a>
      </li>
      <li
        class="navigation__item"
      >
        <a
          class="navigation__link"
          href="/cart"
        >
          Cart
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
